src/lib/logging/Logger.js

```
@@ -0,0 +1,314 @@
+import * as LogLevel from 'loglevel';
+
+import LoggerCache from './LoggerCache';
+import { LOG_CODES } from './logConstants';
+import { arrayToString, sortLogsByTime, printLog } from './logUtils';
+
+/**
+ * Logging mechanism that allows for storage of log messages, saving to backend, and 
+ * controlling what messages are shown to the user via browser console.
+ */
+class Logger {
+    /** @property {Object} - Instance of internal logger */
+    logger;
+
+    /** @property {LoggerCache} - Cache for storing and validating log messages */
+    cache;
+
+    /**
+     * @constructor
+     */
+    constructor() {
+        this.cache = new LoggerCache();
+        this.logger = LogLevel.noConflict();
+
+        this.onUncaughtError = this.onUncaughtError.bind(this);
+        window.addEventListener('error', this.onUncaughtError);
```

If we catch/swallow an error will they not trigger this callback?